torchdynamo/optimizations/python_key.py

```diff
@@ -121,13 +140,63 @@ def unpack(x):
 
         return tuple(unpack(x) for x in out)
 
+    class PythonKeyTracer(Tracer):
```

Comment:
How is this different from `functorch._src.python_key.PythonKeyTracer`?

Can we just extend PythonKeyTracer to take a `python_tensor_cls` arg?

Reply:
Let's avoid copy any pasting it.